packages/view3d/src/plugin/PostProcessing/PostProcessing.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2020 NAVER Corp.
+ * egjs projects are licensed under the MIT license
+ */
+
+import * as THREE from "three";
+import { EffectComposer } from "three/examples/jsm/postprocessing/EffectComposer";
+import { RenderPass } from "three/examples/jsm/postprocessing/RenderPass";
+import { ShaderPass } from "three/examples/jsm/postprocessing/ShaderPass";
+import { GammaCorrectionShader } from "three/examples/jsm/shaders/GammaCorrectionShader";
+
+import View3D from "../../View3D";
+import { getObjectOption } from "../../utils";
+import View3DPlugin from "../View3DPlugin";
+
+import BloomEffect, { BloomOptions } from "./BloomEffect";
+import DoFEffect, { DoFOptions } from "./DoFEffect";
+import SSAOEffect, { SSAOOptions } from "./SSAOEffect";
+import SSREffect, { SSROptions } from "./SSREffect";
+
+
+export interface PostProcessingOptions {
+  SSAO: boolean | Partial<SSAOOptions>;
+  DoF: boolean | Partial<DoFOptions>;
+  SSR: boolean | Partial<SSROptions>;
+  Bloom: boolean | Partial<BloomOptions>;
+}
+
+export interface PostProcessingEffects {
+  ssaoEffect: SSAOEffect;
+  dofEffect: DoFEffect;
+  ssrEffect: SSREffect;
+  bloomEffect: BloomEffect;
+}
+
+class PostProcessing implements View3DPlugin, PostProcessingEffects {
+
+  // Options
+  private _ssao: PostProcessingOptions["SSAO"];
+  private _dof: PostProcessingOptions["DoF"];
+  private _ssr: PostProcessingOptions["SSR"];
+  private _bloom: PostProcessingOptions["Bloom"];
+
+  // Effects
+  private _ssaoEffect: PostProcessingEffects["ssaoEffect"];
+  private _dofEffect: PostProcessingEffects["dofEffect"];
+  private _ssrEffect: PostProcessingEffects["ssrEffect"];
+  private _bloomEffect: PostProcessingEffects["bloomEffect"];
+
+  private _view3D: View3D;
+  private _renderer: THREE.WebGLRenderer;
+  private _composer: EffectComposer;
+
+  public get ssaoEffect() { return this._ssaoEffect; }
+
+  public get dofEffect() { return this._dofEffect; }
+
+  public get ssrEffect() { return this._ssrEffect; }
+
+  public get bloomEffect() { return this._bloomEffect; }
+
+  public get composer() { return this._composer; }
+
+  public constructor({
+    SSR = false,
+    SSAO = false,
+    DoF = false,
+    Bloom = false
+  }: Partial<PostProcessingOptions>) {
```

Comment:
This could be handy, but it seems like it's quite overkill to pack every post-processing effect in one component.
As this will grow further, increasing bundle size with unused effects as we add more effects.
So, here's a small idea:

I like the idea of how you organized those effects as separate `Effects`, as I thought you'd make them as `View3DPlugin`.
So instead of making them as plugins, how about creating `effects` in View3D instead?
Here's a simple demonstration of API:
```ts
import View3D, { BloomEffect, DoFEffect } from "@egjs/view3d";

const viewer = new View3D("#el", { ... });


const bloom = new BloomEffect();

// Add new post-processing effects
viewer.addEffects(bloom, new DoFEffect());

// Remove effects
viewer.removeEffects(bloom);

// Get activated effects (readonly)
viewer.effects;
```

Reply:
Thank you for your great opinion!!

Additional, how about make more simple of api?

Like this:


```javascript
import View3D from "@egjs/view3d";

const viewer = new View3D("#el", { ... });

// Add new post-processing effects
viewer.addEffects({
  bloom: true,
  dof: {...}
});

// Remove effects
viewer.removeEffects('bloom');

// Get activated effects (readonly)
viewer.effects;
```

And use Dynamic Import in View3D for maintain bundle size.

Like this:
```typescript
public addEffects(param: Partial<PostProcessingOptions>){
  if(param.bloom){
    import(...).then(BloomEffect => {
     this.composer.add(new BloomEffect(param.bloom));
    });
  }

  if(param.dof){
    import(...).then(DoFEffect => {
     this.composer.add(new DoFEffect(param.dof));
    });
  }
}
```

Reply:
Just I hope View3D users can use it more easily and comfortably. :)